cdap-proto/src/test/java/co/cask/cdap/proto/codec/NamespacedIdCodecTest.java

```diff
@@ -40,12 +40,12 @@
   private final Id.Namespace ns = Id.Namespace.from("ns");
   private final Id.Application app = Id.Application.from(ns, "app");
   private final Id.Program program = Id.Program.from(app, ProgramType.CUSTOM_ACTION, "action");
-  private final Id.Flow flow = Id.Flow.from(app, "flow");
```

Comment:
This pattern is consistent across this PR. So can't users ever do ``Id.Flow flow = Id.Flow.from(app, "flow")`` now? Does the type always have to be declared as Program? I think that's because ``Id.Flow`` and others don't extend ``Id.Program``. What was the reason for that change?